### DISCUSSION TITLE
Emporia VUE gen 2 stock firmware needed

### DISCUSSION THREAD

**Discussion by danutzgtx480** [emporia-vue-local/esphome#79] [General]
Hello everybody,
Does anyone have the stock .bin firmware for the Emporia VUE gen 2 device ?
I need the firmware as I cannot connect to the device , it just does a long beep every few seconds with the green led on and off and it cannot be found via Bluetooth 
Thank you

**Comment by baylanger** [comment]
Is the serial number/code burned into the firmware? If not, in the coming days I’m planning to extract the original firmware from unboxing.

**Comment by PepereLoulou** [comment]


**Reply by danutzgtx480** [reply]
Thank you,will test and report back in few days.

**Comment by danutzgtx480** [comment]
The firmware flash worked but it was still having the beeping / not detected issue  .I noticed that when it was powered by the 5V VCC pin (not connected to mains ) the device can be paired so I figured it must be a power supply issue .After checking on the back there was a blown resistor,replaced it with a similar one (510 ohm ) and the device is working fine now.Using Esphome firmware for a better Home assistant integration.

**Reply by baylanger** [reply]
Would you have a picture showing the location of that resistor or perhaps it's associated number on the board, Rxx ?? It's surprising to see a resistor blown up.

**Reply by danutzgtx480** [reply]
It's resistor R98 . Didn't have an SMD one ,so I just used a regular one and it looks to work fine.

**Reply by danutzgtx480** [reply]
I have used the following picture of the board to get the value of the SMD resistor as mine was blown and not visible.
https://flaviutamas.com/assets/images/2021-07-28-reversing-emporia-vue-2/back.jpg

**Comment by baylanger** [comment]
My virgin firmware, taken prior to make any config, differs from the one posted by @PepereLoulou - quickly I found these to be different.

```
LE_LOCAL_KEY_IRK = 
LE_LOCAL_KEY_IR = 
LE_LOCAL_KEY_DHK = 
```

They are probably uniq for each device and allows to talk to Emporia cloud?

```
ssid xxALVEZxx
```

@PepereLoulou is that your wifi? If so, consider changing your password :)

**Reply by PepereLoulou** [reply]
Yes it's my wifi ^^

**Reply by baylanger** [reply]
I edited your wifi name hoping it gives you enough time to change all your devices' password 😛

**Reply by PepereLoulou** [reply]
Thanks ;-)

**Comment by pengsloth** [comment]
@PepereLoulou or @baylanger still have your stock firmware? Mine bit the dust when my laptop died and I'd like to revert my Gen 2 as well. There is no longer a link on the above comments.

**Reply by PepereLoulou** [reply]
Yes:

https://drive.google.com/file/d/1Y8GPPdtOhIDtmgnR4qSrPNXVLoyRTLXF

**Reply by pengsloth** [reply]
Thanks!

**Reply by pengsloth** [reply]
Strange... in ESPhome-flasher I'm getting this error:

Unexpected error: The firmware binary is invalid (magic byte=FF, should be E9)

**Reply by senoird** [reply]
That happened to me also. Grounding the IO0 pin while flashing solved it in my case.

**Comment by t0ny-peng** [comment]
I lost my original firmware too. Btw can you simply change the number mentioned by Baylander, e.g., `LE_LOCAL_KEY_***` and flash it back? Thx!

**Comment by jeedee5** [comment]
The google drive is not available anymore... Can anybody share the original firmware?
Not that I don't like the esphome, but I am having trouble getting correct voltages (3-phase+N, so I want to first try to go back to the original and test this again...
I did a lot of test, and to me, it looks like the SAMD09 is stuck in a wrong function (more like US 2x120v instead of European 3-phase+N)

Phase A Voltage | 287.7 V
Phase B Voltage | 349.5 V
Phase C Voltage | 172.0 V
Phase B Phase Angle | 80 °
Phase C Phase Angle | 256 °

**Reply by baylanger** [reply]
Here's a virgin backup of the firmware.

Note: that I edited all of the following. Unless you kept of copy of the original values of your firmware, it will probably prevent your device to connect to Emporia servers... so the firmware below might be useless.

LE_LOCAL_KEY_ER
LE_LOCAL_KEY_IR
LE_LOCAL_KEY_DHK
LE_LOCAL_KEY_IRK

https://www.dropbox.com/scl/fi/76b22ylko5urfa9g4x6ah/Emporia-Original-Virgin-evue2a-no-serial.bin?rlkey=t8fm03z1kne6tcwgudasjdzre&st=xigxvkqi&dl=0

**Reply by jeedee5** [reply]
Virgin firmware worked perfectly, without adding the above keys... Device configures over bluetooth, and connects to the cloud after wifi config. A new firmware was installed automatically. 

but still my problem of the wrong voltages beeing logged is still the same... continuing investigations...

**Reply by ngist** [reply]
> but still my problem of the wrong voltages beeing logged is still the same... continuing investigations...

Make sure your antenna isn't bonded to the panel. It needs to be insulated.
